### DISCUSSION TITLE
Issue with Displaying Multilingual News on SharePoint PNP Modern Search Page

### DISCUSSION THREAD

**Discussion by h3nn3** [microsoft-search/pnp-modern-search#4072] [Q&A]
Hello community :),

I have a SharePoint site with multilingual capabilities activated, and the default language is English. Here is the scenario:

1. I create a news post in English and translate it into German.
2. I create another news post in English but do not translate it into German.

I then create a search page, which is initially in English. From this search page, I create a German version. On the German search page, I want to display all news posts. However, I want the following behavior:

- If a news post has a German translation, only the German version should be displayed.
- If a news post does not have a German translation, the default English version should be displayed.

I have reviewed other posts and tried the following query snippet, but it does not work in my scenario:

`(SPTranslationLanguage:{Page._SPTranslationLanguage} OR (-SPTranslatedLanguages:{Page._SPTranslationLanguage} AND -SPIsTranslation:true))`

**Goal:** The goal is to ensure that the end user sees all news posts on the search page. If a news post is available in German, only the German version should be shown. If a news post does not have a German translation, the default English version should be displayed to ensure no information is missed.

**Request:** Could you please provide guidance or a solution to achieve this functionality?

Thank you!

**Comment by kasperbolarsen** [comment]
The article might be just what you are looking for: https://www.techmikael.com/2020/09/search-and-modern-multilingual-pages-in.html

**Reply by Jeroen00** [reply]
@kasperbolarsen @wobba As Mikael writes in his blog post from 2020, the managed property SPTranslationLanguage should contain the page language (when page translation is used). However, we see in several environments that this property ONLY contains a value for the translated pages, for the original (default language) page this property is always NULL. Since we are getting pages from different sites with different default languages, we do not know what the page language is if the property returns as NULL. Does anyone know if this is 'by design' or a bug that Microsoft should be able to resolve?
